docs/design/uptime-reputation-calculation.md

```diff
@@ -0,0 +1,177 @@
+# Uptime reputation caculation
+
+## Abstract
+
+Having a fair, simple and understandable way to calculate the uptime of _storagenodes (henceforth SNs)_ is crucial for not causing unfair disqualifications, clarifying how _storagenode operators (henceforth SNOs)_ should administrate and maintain their _SN_ for being align with the network requirements and last but not least have a clear proof of disqualification for _SN_ getting disqualified due to too many uptime check failures.
+
+This design doc proposes a different way for calculating uptime check scores.
+
+## Background
+
+Currently our disqualification system is described in the [disqualification design doc](disqualification.md) which is based in the calculation of different _SN_ reputation scores as described in [node-selection design doc](node-selection.md).
+
+The problem with the current uptime _SN_ reputation score calculation is that it isn't based on the actual time that a node is being offline, by cause of to the nature of the chosen model, causing that an _SN_ will get disqualified due its number of uptime checks success and failure which implies that new nodes can get disqualified having less uptime check failures than veteran nodes.
+
+While we understand that an _SN_ that has remained in the network with a low uptime check failures could be considered a more reliable _SN_ than new ones which recently joined, we see, at the same time, that new _SN_ shouldn't be harsher penalized than veteran nodes due to uptime check failures because:
+
+1. Any _SN_ (veteran and new ones) requires some time to be offline for maintenance and both should have the same maximum amount of time for it.
+1. The end goal of the uptime checks is to ensure that the network has certain availability and retrievability and uptime check failures affect to them no matter on how long the _SN_ has remained in the network and how much data an _SN_ has, hence node with more or less data must be treated equally.
+
+This design doc is focused on how to design and implement a fair, simple and understandable uptime reputation calculation after our initial approach didn't behave as expected, causing the disqualification of several nodes without having a proof that they have failed enough uptime checks that the _SNs_ were offline long time enough which put in danger the minimum level of the network availability and retrievability.
```

Comment:
This is a nit pick, but to make this sentence a little easier to read, you could break it up like this...
"This design doc is focused on how to design and implement a fair, simple and understandable uptime reputation calculation after our initial approach didn't behave as expected. Our original design caused the disqualification of several nodes without having a proof that they have failed enough uptime checks. This flaw put the minimum level of the network availability and retrievability at risk."